### PR TITLE
Feature/ab#23663 synchronization part 3

### PR DESCRIPTION
**Pull request by jimmyPasta**
The synchronize now performs a get request to chefs for 2 days of missing requests.
The get method/sycnh does not care if the submission button was not named correctly.

This method hypothetically could be called more often but hopefully it is not needed if we can get past chefs issues of not posting when the submit button is not named correctly.

This synchronization will check to see if the version of the form mapping exists. 
If it does not it will try to create  the default mapping and then will inform the notifications channel that the version was initialized and that the custom mapping needs doing.

The next day that this is run and the default version mapping exists it will do the synchronization which I don't know if it is the best approach.

Discussed with Stephan, for now we are ok with pushing through submissions with default mapping.
Because of our due dilligence on verifying that for every submision has a mapping right now this is not an issue.
It will become an issue when we have multiple onboards and we can not control the flow

### CODE REVIEW

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/Intakes/BackgroundWorkers/IntakeSyncWorker.cs [hunk 1]
```diff
@@ -39,12 +39,12 @@ public override async Task Execute(IJobExecutionContext context)
             Logger.LogInformation("Executing IntakeSyncWorker...");
 
             var tenants = await _tenantRepository.GetListAsync();
-
+            var numberDaysBack = -2;
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Application/Intakes/BackgroundWorkers/IntakeSyncWorker.cs:42]: should we make this part of the BackgroundJobsOptions config?

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormSycnronizationService.cs [hunk 2]
```diff
@@ -34,105 +37,145 @@ public class ApplicationFormSycnronizationService :
...
                         HashSet<string> newChefsSubmissions = await GetChefsSubmissions(applicationFormDto, numberOfDaysToCheck);
                         HashSet<string> existingSubmissions = GetSubmissionsByForm(applicationFormDto.Id);
                         missingSubmissions = newChefsSubmissions.Except(existingSubmissions).ToHashSet();
-                        Logger.LogInformation("Appform SYNC: Looking up missing: " + missingSubmissions);
+
+                        AddFact("------------------------------------", "----------------------------------------");
+                        AddFact("Application Form Name: ", applicationFormDto.ApplicationFormName);
+                        AddFact("Missing Submissions Count: ", missingSubmissions.Count.ToString());
+
                         if (missingSubmissions != null && missingSubmissions.Count > 0)
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormSycnronizationService.cs:167]: missingSubmissions can never be null, so dont think this check is necessary

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormSycnronizationService.cs
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormSycnronizationService.cs:1]: line 266: CA2254 warning

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormVersionAppService.cs [hunk 3]
```diff
@@ -59,31 +60,41 @@ public override async Task<ApplicationFormVersionDto> GetAsync(Guid id)
             return dto;
         }
 
-        public async Task<bool> InitializePublishedFormVersion(dynamic chefsForm, Guid applicationFormId)
+        private static JToken? GetFormVersionToken(dynamic chefsForm) {
+            if (chefsForm == null) return null;
+            JObject formObject = JObject.Parse(chefsForm.ToString());
+            if (formObject == null) return null;
+
+#pragma warning disable CS8600
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormVersionAppService.cs:68]: can remove this by marking the JToken versionsToken as JToken? versionsToken

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/Intakes/BackgroundWorkers/IntakeSyncWorker.cs (hunk 1 above)
Comment (jimmyPasta) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/Intakes/BackgroundWorkers/IntakeSyncWorker.cs:42]: Yeah probably though it seems to be a bit of a pain getting config options in

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormSycnronizationService.cs [hunk 4]
```diff
@@ -34,105 +37,145 @@ public class ApplicationFormSycnronizationService :
...
         private readonly IApplicationFormRepository _applicationFormRepository;
+        private readonly ICurrentTenant _currentTenant;
         private readonly IApplicationFormSubmissionRepository _applicationFormSubmissionRepository;
-        private readonly IChefsMissedSubmissionsRepository _chefsMissedSubmissionsRepository;
+        private readonly IApplicationFormVersionAppService _applicationFormVersionAppService;
         private readonly IConfiguration _configuration;
+        private readonly ISubmissionsApiService _submissionsApiService;
+        private readonly IIntakeFormSubmissionManager _intakeFormSubmissionManager;
+        private List<Fact> _facts = new();
+        private HashSet<string> _formVersionsInitializedVersionHash = new HashSet<string>();
```
[the remark this answers is not in the document]
Comment (jimmyPasta) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormSycnronizationService.cs]: Yes, it is now removed

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormVersionAppService.cs (hunk 3 above)
Comment (jimmyPasta) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormVersionAppService.cs]: Removed - fixed

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormSycnronizationService.cs (hunk 2 above)
Comment (jimmyPasta) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/ApplicationForms/ApplicationFormSycnronizationService.cs]: Removed - fixed

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/Intakes/BackgroundWorkers/IntakeSyncWorker.cs (hunk 1 above)
Comment (jimmyPasta) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/Intakes/BackgroundWorkers/IntakeSyncWorker.cs:42]: Going to setup a config param

**Review by AndreGAot — APPROVED** [review]
Ready for Dev

**Pull request merged**